### ISSUE TITLE
lldb in OSX cannot continue execute after catch BAD_ACCESS

### ISSUE THREAD

**Issue by HerrCai0907** [llvm/llvm-project#60438]
I try to use lldb in macos to debug a program with self-defined SIGBUS handler. But lldb cannot handle the EXC_BAD_ACCESS. After catch this error, it cannot continue execute anymore.

system version:
macOS Ventura 13.2

https://stackoverflow.com/a/32724035 can solve this issue but it need to patch something and rebuild lldb.
It there any more easier way to do it?

**Comment by HerrCai0907** [comment]
A work around solution is using environment `LLDB_DEBUGSERVER_EXTRA_ARG_1=--unmask-signals`
It will pass additional command line argument `--unmask-signals` to debugserver. And debugserver will call following function and set mask for `EXC_MASK_BAD_ACCESS`.

```c++
void RNBContext::AddDefaultIgnoredExceptions() {
  m_ignored_exceptions.push_back(EXC_MASK_BAD_ACCESS);
  m_ignored_exceptions.push_back(EXC_MASK_BAD_INSTRUCTION);
  m_ignored_exceptions.push_back(EXC_MASK_ARITHMETIC);
}
```

Could it possible to enable this feature in OSX default?  I am happy to prepare a patch for it.

**Comment by jimingham** [comment]
If you are using a recent lldb, you should have the setting: platform.plugin.darwin.ignored-exceptions which you can use to control which exceptions are unmasked.

You get more information from the EXC_BAD_ACCESS than you do from the SIGBUS, and very few programs have SIGBUS handlers, so I think having the default be to catch the exception rather than letting it always propagate to the SIGBUS is the right choice.  And with this setting you have more control over which exceptions you want to have unmasked.

Jim

**Comment by HerrCai0907** [comment]
Both `settings set platform.plugin.darwin.ignored-exceptions EXC_BAD_ACCESS` and `export LLDB_DEBUGSERVER_EXTRA_ARG_1=--unmask-signals` can solve it.

**Comment by icemmw** [comment]
> 
I read the source code, and I definitely believe you are right! But when I tried type the command in lldb, I got this error: invalid value path 'platform.plugin.darwin.ignored-exceptions', I don't know how to fix that. Pls tell me how to do that?  
Thx a lot sincerely!

**Comment by DavidSpickett** [comment]
You may need to update your lldb, I get this from the latest build:
```
$ ./bin/lldb
(lldb) settings set platform.plugin.darwin.ignored-exceptions EXC_BAD_ACCESS
```
And I can get the error you're seeing by deliberately messing up the name:
```
(lldb) settings set platform.plugin.darwin.ignored-exceptionszzzz EXC_BAD_ACCESS
error: invalid value path 'platform.plugin.darwin.ignored-exceptionszzzz'
```

Let me see if it's obvious when this setting was added.

**Comment by DavidSpickett** [comment]
It was added by https://github.com/llvm/llvm-project/commit/bff4673b41781ec5bff6b96b52cf321d2271726c and this is first present in the 15.x releases, go with 15.0.7 if you can.

How that translates into XCode lldb versions, I don't know, but perhaps you can get it from Homebrew if you were previously using the system lldb.

**Comment by icemmw** [comment]
> It was added by [bff4673](https://github.com/llvm/llvm-project/commit/bff4673b41781ec5bff6b96b52cf321d2271726c) and this is first present in the 15.x releases, go with 15.0.7 if you can.
> 
> How that translates into XCode lldb versions, I don't know, but perhaps you can get it from Homebrew if you were previously using the system lldb.

Thx for your answer! Yes, it is installed by XCode-select, and my XCode version is 12.4. After several times I thought the reason is lldb-version is not compatible, so last night, and I build lldb by source code quickly. Now I can type this command, but I'm not sure if it worked fine, because the issue is still right there.

The example as below:
<img width="768" alt="lldb_not_pass" src="https://github.com/llvm/llvm-project/assets/14175652/79aea7dc-e7cc-4191-a6ee-d7d05267bd2e">

I really don't know what happened, but it presented un-normal when I tried use lldb, I always got exception as I said before. In the opposite side, it is launched as usual when I double-clicked the app or I type "open xxxx.app" in terminal.

**Comment by Petia** [comment]
I tried `settings set platform.plugin.darwin.ignored-exceptions EXC_BAD_ACCESS` on my lldb (latest version, just updated to XCode 14.3.1, so lldb-1403.0.17.67). Surprisingly, it didn't change a thing: lldb is still stuck on BAD_ACCESS exceptions. Any advice or diagnosis?

**Comment by jimingham** [comment]
This setting should cause debugserver to "mask out EXC_BAD_ACCESS" from the set of exceptions it is waiting for.  So if you see EXC_BAD_ACCESS at all, it means the setting isn't working for some reason.  Note, this setting causes lldb to send a packet to debugserver, telling it to mask out this exception, but debugserver also has to support this packet, which was added on both lldb & debugserver ends at the same time.  So you have to make sure both that your lldb version supports sending the packed and debugserver supports receiving it.

Jim

**Comment by Petia** [comment]
Today I just updated to XCode 15, so I now have lldb-1500.0.22.8. After having chased debugserver on my drive, I found it on a dark corner of the LLDB.Framework; running it -version gives "debugserver-@(#)PROGRAM:LLDB  PROJECT:lldb-1500.0.22.8", so, that seems the corresponding version.
Nevertheless, LLDB still stops at EXC_BAD_ACCESS no matter what. This is a is very frustrating situation, since it prevents me from debugging any code that calls Java (because of such an exception in the Java graphics engine).

This not a config issue:
```
(lldb) settings show platform.plugin.darwin.ignored-exceptions
    platform.plugin.darwin.ignored-exceptions (string) = "EXC_BAD_ACCESS"

```
Any suggestion welcome.

[FWIW, Macbook Pro M1, Ventura 13.5.2]
